uefi/src/table/runtime.rs

```diff
@@ -651,3 +714,52 @@ pub struct CapsuleInfo {
     /// The type of reset required for the capsule update.
     pub reset_type: ResetType,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use alloc::string::ToString;
+    use core::slice;
+
+    unsafe fn time_as_u8_slice(p: &Time) -> &[u8] {
+        slice::from_raw_parts(core::ptr::addr_of!(*p).cast(), size_of::<Time>())
+    }
+
+    #[test]
+    fn test_time_from_bytes() {
+        let mut time;
+        let mut time_from_bytes;
+        let mut time_params = TimeParams {
+            year: 2024,
+            month: 6,
+            day: 13,
+            hour: 4,
+            minute: 29,
+            second: 30,
+            nanosecond: 123_456_789,
+            time_zone: None,
+            daylight: Daylight::empty(),
+        };
+
+        time = Time::new(time_params).unwrap();
+        unsafe {
+            time_from_bytes = Time::try_from(time_as_u8_slice(&time)).unwrap();
+        }
+        assert_eq!(time, time_from_bytes);
+
+        time_params.time_zone = Some(120);
+        time = Time::new(time_params).unwrap();
+        unsafe {
+            time_from_bytes = Time::try_from(time_as_u8_slice(&time)).unwrap();
+        }
+        assert_eq!(time.to_string(), time_from_bytes.to_string());
+
+        time_params.time_zone = Some(150);
+        time = Time::new(time_params).unwrap();
+        unsafe {
+            time_from_bytes = Time::try_from(time_as_u8_slice(&time)).unwrap();
+        }
+        assert_eq!(time.to_string(), time_from_bytes.to_string());
```

Comment:
I think these are all successful cases? Add at least one `InvalidFields` test and one `InvalidSize` test.